hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -999,6 +1006,8 @@ public AbfsRestOperation read(final String path, final long position, final byte
             bufferLength, sasTokenForReuse);
     op.execute(tracingContext);
 
+    verifyCheckSumForRead(buffer, op.getResult());
```

Comment:
Good that we are taking dataRead from op. But we need to send the offset so that array copy take correct bytes.

Probably lets add test for this. We can write 8 MB data and read it
```
inputStream.read(byte_array_8_mb, 8MB);
```